src/libraries/System.Private.Uri/src/System/Uri.cs

```diff
@@ -1128,7 +1109,7 @@ public string DnsSafeHost
 
                 EnsureHostString(false);
 
-                if (!string.IsNullOrEmpty(_info.DnsSafeHost))
+                if (_info.DnsSafeHost != null)
```

Comment:
Why is this change ok... it's never empty if it's not null?  Does that need to be asserted below before we return?